### PR TITLE
[ADD]: custom collections permissioning support

### PR DESCRIPTION
**Pull request by madisonmay**
Reviewers: 
----------

- [ ] @madisonmay
- [ ] @sihrc 
- [ ] @anneCarlson 

What's the purpose of the PR?
-----------------------------

Add support for custom collection's permissioning.

Description of changes
----------------------

Add support for custom collection's permissioning.

Notes to reviewers
------------------

Keep in mind while reviewing code:
----------------------------------
- Is relevant code tested?
- Are added functions/methods documented?
- Separation of concerns (SOC)
- Don't repeat yourself (DRY) 
- Limit the number of positional arguments
- Is function/method length reasonable?
- Can code be broken down into smaller components?
- Where do added functions/methods belong?
- Are variable names descriptive?
- Are errors handled appropriately?
- Can logic be simplified?
- Does the code make sense in context? (expand the diff)

### CODE REVIEW

File: R/collection.R [hunk 1]
```diff
@@ -157,6 +157,148 @@ setMethod(f="info",
...
+          signature="Collection",
+          definition=function(collection_object, name, version = NULL, ...) {
+            previous_name <- collection_object@name
+            if (!is.null(name)) {
+              collection_object@name <- name
+            } else {
+              stop("NULL is not a valid collection name.")
+            }
+            make_request(NULL, 'custom', version = version, collection = previous_name, name = collection_object@name, method = "rename", ...)
+            collection_object
```
Comment (anneCarlson) [inline R/collection.R]: We need to make sure this is documented well in the docs.

File: tests/testthat/test-collection.r [hunk 2]
```diff
@@ -85,3 +89,76 @@ test_that("instantiate a collection and clear it", {
   expect_true(is.null(collections[[test_string]]))
   clearTests(collection)
 })
+
```
Comment (anneCarlson) [inline tests/testthat/test-collection.r]: We need a rename test.

Comment (madisonmay) [inline tests/testthat/test-collection.r]: test below?

Comment (madisonmay) [inline tests/testthat/test-collection.r]: the two info tests check that the local name of the collection changes, and that new collections instantiated with the new name also access the collection

File: R/api_handler.R [hunk 3]
```diff
@@ -45,15 +47,13 @@ make_request <- function(data, api, version = NULL, apis = NULL, method = NULL,
...
-  stop_for_status(response)
-
-
   # Returns results
   answer <- content(response, as = "parsed", type = "application/json")
   if ("error" %in% names(answer)) {
     stop(answer[["error"]])
   }
 
+  stop_for_status(response)
```
Comment (sihrc) [inline R/api_handler.R]: why did we move this down? I assume it waits for response to finish up.

Comment (madisonmay) [inline R/api_handler.R]: previously we were error with code 400 rather than erroring with the error message indicoAPI wanted us to raise

Comment (madisonmay) [inline R/api_handler.R]: so this makes errors more specific

**Pull request merged**